src/compiler/diagnosticMessages.json

```
@@ -6516,6 +6516,10 @@
         "category": "Error",
         "code": 8016
     },
+    "Signature declarations can only be used in TypeScript files.": {

```

@RyanCavanaugh what do you think of just making this

```suggestion
    "This function is missing a body.": {
```

---

IMO we should point out that it's only invalid in JS files because it's a TS proprietary syntax. The other error messages all use the same format.

---

I agree with @Josh-Cena. It seems more likely that someone copies+pastes an overloaded function from a TS file to a JS file than that they happen to accidently drop a semicolon in that exact position

---

I think you're right that's a very rare mistake, **but** anybody who does is likely to be incensed that VSCode thinks they're using Typescript -- based on my experience of other do-not-use-typescript-in-JS errors.

I'd still go with the error as it is in the PR, though. @DanielRosenwasser do you agree?